src/utils/messenger.h

```diff
@@ -176,16 +201,21 @@ class Messenger {
 private:
     int size;
     int rank;
+    int color;
```

Comment:
I don't know if color is a common concept. Is it easy to understand for others? add some comment?

Reply:
Yes, color is common concept from https://www.mpich.org/static/docs/v3.1.3/www3/MPI_Comm_split.html.